cypress/e2e/Taxdata/Pricing.cy.js

```diff
@@ -0,0 +1,108 @@
+describe('Pricing Page Test Suite', () => {
+    Cypress.on('uncaught:exception', (err, runnable) => {
+        // Preventing Cypress from failing the test on uncaught exceptions
+        return false;
+    });
+
+    before(() => {
+        cy.visit('https://apilayer.com/marketplace/tax_data-api');
+    });
+
+    it('1. Test to accept Cookies if not already accepted', () => {
+        cy.AcceptCookies();
+    });
+
+    it('2. Test to check for the "Pricing" tab present or not', () => {
+        cy.componentVisiblityCheck('#pricing-tab > .d-md-flex > span');
+    });
+
+    it('3. Test to click on the "Pricing" tab', () => {
+        cy.get('#pricing-tab > .d-md-flex > span').click();
+    });
+
+    it('4. Test to check for "Free Plan" present or not', () => {
+        cy.componentVisiblityCheck(':nth-child(1) > .card > .card-header > .h3', 'Free Plan');
+    });
+
+    it('5. Test to check "Free Plan" price is $0.00', () => {
+        cy.componentVisiblityCheck(':nth-child(1) > .card > .card-header > .mb-3 > .plan-price','$0');
+    });
+
+    it('6. Test to check "Free Plan" features (100 requests, free for lifetime, no credit card)', () => {
+        cy.componentVisiblityCheck('.media-body');
+        cy.componentVisiblityCheck(':nth-child(1) > .card > .card-body > :nth-child(2) > .media-body', 'Free for Lifetime');
+        cy.componentVisiblityCheck(':nth-child(3) > .media-body', 'No Credit Card Required');
+    });
+
+    it('7. Test to check "Starter Plan" present or not', () => {
+        cy.componentVisiblityCheck(':nth-child(2) > .card > .card-header > .h3','Starter Plan');
+    });
+
+    it('8. Test to check "Starter Plan" price is $14.99', () => {
+        cy.componentVisiblityCheck(':nth-child(2) > .card > .card-header > .mb-3 > .plan-price', '$19.99');
+    });
+
+    it('9. Test to check "Starter Plan" features (10,000 requests, standard support)', () => {
+        cy.componentVisiblityCheck(':nth-child(2) > .card > .card-body > :nth-child(2) > .media-body', 'Standard Support');
+        cy.checkTextVisibility('1,000 Requests / Monthly');
+    });
+
+    it('10. Test to check "MOST POPULAR" label on Starter Plan', () => {
+        cy.componentVisiblityCheck('.card > .badge','MOST POPULAR');
+    });
+
+    it('11. Test to check "Pro Plan" present or not', () => {
+        cy.componentVisiblityCheck(':nth-child(3) > .card > .card-header > .h3', 'Pro Plan');
+    });
+
+    it('12. Test to check "Pro Plan" price is $49.99', () => {
+        cy.componentVisiblityCheck(':nth-child(3) > .card > .card-header > .mb-3 > .plan-price', '$49.99');
```

Comment:
this sort of hierarchy will never change?

Reply:
All the test cases involve this type of approach to get the element. You have the class name cant you get the element using classname?

Reply:
didn't  find the classname for this and from cypress selector only we taken this,and there is no other way to get the elements.